### ISSUE TITLE
Please update xmlf90 to the latest version

### ISSUE THREAD

**Issue by yurivict** [dftbplus/dftbplus#171]
The latest version is 1.5.4: https://launchpad.net/xmlf90/

It doesn't install ```include/m_strings.mod``` that ```dftbplus``` expects.
I have the ```xmlf90``` package installed, and would rather use it instead of the bundled version.

There is a policy on FreeBSD that external libs should be used instead of the bundled ones.
The reason for this policy is that if security problems are found in the libs that are bundled, it is much harder to find/fix them.

Thank you,
Yuri

**Comment by aradi** [comment]
Dear @yurivict, I understand and FreeBSD policy, and I think, this is a resonable idea. However, our xmlf90 somewhat diverted from the original one:

* the original xmlf90 library had a few issues, which I have fixed during integration. 

* If I remember correctly, I also added a few extensions to speed it up

* and finally I rewrote the string module to avoid pointers and memory leaks resulting when forgetting to call unstring() somewhere.

As at that time the xmlf90 project was officially declared dead (Alberto Garcia diverted me to FoX, when I have sent him my patches), I did not care about the project any more.

I definitely do not want to reintroduce the pointer based strings and the unstring() calls in DFTB+ again, so if somebody volunteers to merge our changes into the official xmlf90, and tests then DFTB+ with that version (and makes sure, it does not have any memory leaks), then we would be more than happy to accept an according pull request. Unfortunately, at the moment I do not have the capacity for doing it myself. Also, I am still somewhat unsure about the status of the xmlf90 project. Is it actively developed now?
